servers/jax-rs-tests/src/test/java/org/projectnessie/jaxrs/TestJerseyRestNaiveClientInMemory.java

```diff
@@ -67,8 +66,8 @@ public NessieClientBuilder<?> configure(NessieClientBuilder<?> builder) {
         (HttpAuthentication) client -> client.addRequestFilter(noAcceptFilter));
   }
 
-  @AfterEach
-  void ensureHeadersProcessed() {
+  @AfterAll
```

Comment:
It does break the original intention of this check, but with API v2 some test cases are suppressed by assumptions, but `@AfterEach` still runs for them and would fail. I'll probably inline the calls to `ensureHeadersProcessed`. Hope it will be clearer.

Reply:
I've changed to annotation based filtering in the latest commit in this PR. I'll probably remove this commit and rebase on top of #5436 when it is merged - just wanted to move the test infra changes into a smaller standalone PR.